docs/state_machine.mmd

```diff
@@ -18,7 +18,10 @@ ClusterStarting --> Savepointing
 ClusterStarting -- Create fails --> DeployFailed
 
 Savepointing --> SubmittingJob
-Savepointing -- Savepoint fails and no externalized checkpoint --> DeployFailed
+Savepointing -- Savepoint fails --> Recovering
+
+Recovering --> SubmittingJob
```

Comment:
Nit:

You might have put more thought into this ? `Recovering` is very generic, and explains the step. But since this is tied to Savepoint now, do you want something specific?  Or are you planning to use this state for other use cases.

Reply:
"Recover" is what we call this currently in our legacy platform. Happy to hear other suggestions though.